### PR TITLE
[PlayLists] Migrate most playlist implementations to tinyxml2

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
This migrates most playlist implementations to tinyxml2.
Implements tests based on example playlists in the code already to verify before/after migration
Split out PlayListRAM and PlayListASX from PlayListPLS into their own files. Gives easier way to test/distinguish the 3 types.

The remaining playlist to migrate to tinyxml2 is PlayistASX. the asx file format is not case sensitive, whereas xml is, and in particular tinyxml2 is case sensitive. Its not as easy to modify things with tinyxml2, but should have something in the not too distant future.

## Motivation and context
Tinyxml2

## How has this been tested?
Tests

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/playlists/PlayListB4S.cpp [hunk 1]
```diff
@@ -50,26 +50,33 @@ CPlayListB4S::~CPlayListB4S(void) = default;
 
 bool CPlayListB4S::LoadData(std::istream& stream)
 {
-  CXBMCTinyXML xmlDoc;
+  CXBMCTinyXML2 xmlDoc;
 
-  stream >> xmlDoc;
+  std::string b4sstream(std::istreambuf_iterator<char>(stream), {});
```
Comment (garbear) [inline xbmc/playlists/PlayListB4S.cpp:55]: For future, camelcase (`b4sStream`) is intended to improve readability. But such a small change, no need to update this PR.

File: xbmc/playlists/PlayListRAM.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#include <iostream>
+
+namespace KODI::PLAYLIST
+{
+class CPlayListRAM : public CPlayList
+{
+public:
+  bool LoadData(std::istream& stream) override;
+};
+} // namespace KODI::PLAYLIST
```
Comment (garbear) [inline xbmc/playlists/PlayListRAM.h:22]: Missing trailing newline. Some ancient C compilers break without a newline (not a concern for us). As a result of these legacy limitations though, many devs have IDEs that automatically insert trailing newlines, so saving this file would dirty their git history. Missing newlines show up best in GitHub diffs so best to scan for this red flag after opening a PR.

**Review by garbear — APPROVED** [review]
The ASX and RAM playlist splits look like they were done with surgical precision, recognizing the need to account for different function visibility. I especially like that you added test cases beyond the existing ones. Good work!

File: xbmc/playlists/PlayListB4S.cpp (hunk 1 above)
Comment (fuzzard) [reply xbmc/playlists/PlayListB4S.cpp]: fixed, for b4s as well as asx and wpl

File: xbmc/playlists/PlayListRAM.h (hunk 2 above)
Comment (fuzzard) [reply xbmc/playlists/PlayListRAM.h]: fixed

**Pull request merged**